packages/web3-eth-ens/src/registry.ts

```diff
@@ -0,0 +1,182 @@
+				.send(txConfig);
+			return receipt;
+		} catch (error) {
+			throw new Error(); // TODO: TransactionRevertError Needs to be added after web3-eth call method is implemented
+		}
+	}
+
+	public async setSubnodeRecord(
+		name: string,
+		label: string,
+		owner: Address,
+		resolver: Address,
+		ttl: number,
+		txConfig: NonPayableCallOptions, // TODO: web3-eth txconfig should be replaced with sendTransaction type
+	) {
+		const hexStrictLabel = !isHexStrict(label) ? sha3Raw(label) : label;
+		try {
+			const receipt = this.contract.methods
+				.setSubnodeRecord(
+					namehash(name),
+					hexStrictLabel,
+					inputAddressFormatter(owner),
+					inputAddressFormatter(resolver),
+					ttl,
+				)
+				.send(txConfig);
+			return receipt;
+		} catch (error) {
+			throw new Error(); // TODO: web3-eth txconfig should be replaced with sendTransaction type
+		}
+	}
+
+	public async setApprovalForAll(
+		operator: string,
+		approved: boolean,
+		txConfig: NonPayableCallOptions, // TODO: web3-eth txconfig should be replaced with sendTransaction type
+	) {
+		try {
+			const receipt = this.contract.methods
+				.setApprovalForAll(operator, approved)
+				.send(txConfig);
+
+			return receipt;
+		} catch (error) {
+			throw new Error(); // TODO: web3-eth txconfig should be replaced with sendTransaction type
+		}
+	}
+
+	public async isApprovedForAll(owner: Address, operator: Address) {
+		try {
+			const promise = await this.contract.methods
+				.isApprovedForAll(inputAddressFormatter(owner), inputAddressFormatter(operator))
+				.call();
+
+			return promise;
+		} catch (error) {
+			throw new Error(); // TODO: TransactionRevertError Needs to be added after web3-eth call method is implemented
+		}
+	}
+
+	public async recordExists(name: string) {
+		try {
+			const promise = await this.contract.methods.recordExists(namehash(name)).call();
+
+			return promise;
+		} catch (error) {
+			throw new Error(); // TODO: TransactionRevertError Needs to be added after web3-eth call method is implemented
+		}
+	}
+
+	public async getResolver(name: string) {
+		try {
+			const promise = await this.contract.methods
+				.resolver(namehash(name))
+				.call()
+				.then(address => {
+					// address type is unknown, not sure why
+					if (typeof address === 'string') {
+						const contract = new Contract(RESOLVER, address);
+						// TODO: set contract provider needs to be added when ens current provider
+						return contract;
+					}
+					throw new Error();
+				});
+
+			return promise;
+		} catch (error) {
+			throw new Error(); // TODO: TransactionRevertError Needs to be added after web3-eth call method is implemented
+		}
+	}
+
+	public async setResolver(
+		name: string,
+		address: Address,
+		txConfig: NonPayableCallOptions, // TODO: web3-eth txconfig should be replaced with sendTransaction type
+	): Promise<ReceiptInfo> {
+		try {
+			const promise = await this.contract.methods
```

Comment:
```suggestion
			return this.contract.methods
```

Good Work Alex, 

Last item: in all set functions, I think we shouldn't wait for promise to resolve in web3.js. and we should return promise and not result, and let user wait for that promise to resolve in his app/lib?

(in that way user will be able to leverage async performance)

Reply:
a part from that all other things lgtm

Reply:
Sounds good, I believe same with the other functions. they should return promises so i took away the awaits